File: 2021/1/main_test.go
package main

import (
	"math"
	"testing"
)

func TestCountIncreases(t *testing.T) {
	sampleInput := importNumsFromFile("testinput.txt")
	increases := countIncreases(sampleInput)
	if increases != 7 {
		t.Errorf("Incorrect Increase Count; got %d, wanted %d", increases, 7)
	}
}

func TestCountWindowIncreases(t *testing.T) {
	sampleInput := importNumsFromFile("testinput.txt")
	increases := countIncreasesWindow(sampleInput, math.MaxInt32)
	if increases != 5 {
		t.Errorf("Incorrect Increase Count; got %d, wanted %d", increases, 5)
	}
}

File: 2021/1/main_test.py
import unittest
import main

class TestCountIncreases(unittest.TestCase):
    testInput = [line.rstrip() for line in open("testinput.txt")]
    def test_count_increases(self):
        count = main.countIncreases(self.testInput)
        self.assertEqual(count, 7, "Wrong Count")
    
    def test_count_increases_window(self):
        count = main.countIncreasesWindows(self.testInput, 3)
        self.assertEqual(count, 5, "Wrong Count")



if __name__ == '__main__':
    unittest.main()